src/utils/getApplicationServerSideProps.js

```diff
@@ -88,6 +88,10 @@ const getApplicationServerSideProps = (callbackFn) => async ({
     };
   }
 
+  // set token in req.headers.cookie so that the token is known when prefetching a request
+  cookies.set('token', query.jwt);
```

Comment:
Shouldn't we add some checks here? if there is a token in the URL and it's different than the one we currently have in the cookie?